src/NServiceBus.Core/EndpointConfiguration.cs

```diff
@@ -34,6 +34,7 @@ public EndpointConfiguration(string endpointName)
             pipelineCollection = new PipelineConfiguration();
             Settings.Set<PipelineConfiguration>(pipelineCollection);
             Settings.Set<SatelliteDefinitions>(new SatelliteDefinitions());
+            Settings.Set<StartupDiagnosticEntries>(new StartupDiagnosticEntries());
```

Comment:
if you would use `GetOrCreate` when specifying sections, this line wouldn't be necessary.

Reply:
`AddStartupDiagnosticsSection` is using `ReadOnlySettings` since it needs to be callable from features, so `GetOrCreate` isn't available.